clientv3/lease.go

```diff
@@ -430,10 +413,9 @@ func (l *lessor) recvKeepAliveLoop() (gerr error) {
 			if canceledByCaller(l.stopCtx, err) {
 				return err
 			}
-		} else {
+		} else if stream != nil {
```

Comment:
why this is changed?

Reply:
I was following the pattern here

```go
ws, err = w.remote.Watch(w.ctx, grpc.FailFast(false)); ws != nil && err == nil {
```

where it's possible `ws==nil && err == nil`.

Do you remember why we had this in the first place? In other retry PR, I have tests failing (`nil pointer`) from `stream == nil && err == nil`. Will investigate more.

Reply:
i do not want to mix fixes together into one commit. this pr should only focus on removing the failfast option and use the shared retry func.

Reply:
Agree. Will revert this change.